acceptance/tests/connect/connect_proxy_lifecycle_test.go

```diff
@@ -205,3 +205,167 @@ func TestConnectInject_ProxyLifecycleShutdown(t *testing.T) {
 		})
 	}
 }
+
+func TestConnectInject_ProxyLifecycleShutdownJob(t *testing.T) {
+	cfg := suite.Config()
+
+	defaultGracePeriod := 10
+
+	cases := map[string]int{
+		"../fixtures/cases/job-client-inject":                  defaultGracePeriod,
+		"../fixtures/cases/job-client-inject-grace-period-0s":  0,
+		"../fixtures/cases/job-client-inject-grace-period-30s": 30,
+	}
+
+	// Set up the installation and static-server once.
+	ctx := suite.Environment().DefaultContext(t)
+	releaseName := helpers.RandomName()
+
+	connHelper := connhelper.ConnectHelper{
+		ClusterKind: consul.Helm,
+		ReleaseName: releaseName,
+		Ctx:         ctx,
+		Cfg:         cfg,
+		HelmValues: map[string]string{
+			"connectInject.sidecarProxy.lifecycle.defaultShutdownGracePeriodSeconds": strconv.FormatInt(int64(defaultGracePeriod), 10),
+			"connectInject.sidecarProxy.lifecycle.defaultEnabled":                    strconv.FormatBool(true),
+		},
+	}
+
+	connHelper.Setup(t)
+	connHelper.Install(t)
+	connHelper.DeployServer(t)
+
+	logger.Log(t, "waiting for static-server to be registered with Consul")
+	retry.RunWith(&retry.Timer{Timeout: 300 * time.Second, Wait: 5 * time.Second}, t, func(r *retry.R) {
+		for _, name := range []string{
+			"static-server",
+			"static-server-sidecar-proxy",
+		} {
+			logger.Logf(t, "checking for %s service in Consul catalog", name)
+			instances, _, err := connHelper.ConsulClient.Catalog().Service(name, "", nil)
+			r.Check(err)
+
+			if len(instances) != 1 {
+				r.Errorf("expected 1 instance of %s", name)
+
+			}
+		}
+	})
+
+	// Iterate over the Job cases and test connection.
+	for path, gracePeriod := range cases {
+		connHelper.DeployJob(t, path) // Default case.
+
+		logger.Log(t, "waiting for job-client to be registered with Consul")
+		retry.RunWith(&retry.Timer{Timeout: 300 * time.Second, Wait: 5 * time.Second}, t, func(r *retry.R) {
+			for _, name := range []string{
+				"job-client",
+				"job-client-sidecar-proxy",
+			} {
+				logger.Logf(t, "checking for %s service in Consul catalog", name)
+				instances, _, err := connHelper.ConsulClient.Catalog().Service(name, "", nil)
+				r.Check(err)
+
+				if len(instances) != 1 {
+					r.Errorf("expected 1 instance of %s", name)
+				}
+			}
+		})
+
+		connHelper.TestConnectionSuccess(t, connhelper.JobName)
+
+		// Get job-client pod name
+		ns := ctx.KubectlOptions(t).Namespace
+		pods, err := ctx.KubernetesClient(t).CoreV1().Pods(ns).List(
+			context.Background(),
+			metav1.ListOptions{
+				LabelSelector: "app=job-client",
+			},
+		)
+		require.NoError(t, err)
+		require.Len(t, pods.Items, 1)
+		jobName := pods.Items[0].Name
+
```

Comment:
I think you might want to check to see if the job can talk to the server before you do the graceful shutdown. That way
you know the proxy is working.